File: KingdomDeathAPI/Models/Settlement.cs
using Swashbuckle.AspNetCore.Annotations;

public class Settlement {
    [SwaggerSchema(ReadOnly = true)]
    public int Id { get; set; }
    public required string Name { get; set; }
    public ICollection<SettlementGearStorageItem> GearStorage { get; set; } = [];
    public ICollection<SettlementResourceStorageItem> ResourceStorage { get; set; } = [];
}

File: KingdomDeathClient/src/app/model/settlement.ts
import { SettlementGearStorageItem } from "./settlement-gear-storage-item";
import { SettlementResourceStorageItem } from "./settlement-resource-storage-item";

export class Settlement {
    id!: number;
    name!: string;
    gearStorage: SettlementGearStorageItem[] = [];
    resourceStorage: SettlementResourceStorageItem[] = [];
}